charts/spire/charts/spire-server/templates/issuer.yaml

```diff
@@ -0,0 +1,47 @@
+{{- with .Values.upstreamAuthority.certManager }}
+{{ if and .enabled .createCA }}
+{{/*
+Configuring CA Issuer: https://cert-manager.io/docs/configuration/ca/
+*/}}
+apiVersion: cert-manager.io/v1
+kind: Issuer
+metadata:
+  name: {{ include "spire-server.fullname" $ }}-signed
+  namespace: {{ include "spire-server.namespace" $ }}
+  labels:
+    {{- include "spire-server.labels" $ | nindent 4}}
+spec:
+{{- if .signedSpec }}
+  {{- toYaml .signedSpec | nindent 4 }}
+{{- else }}
+  selfSigned: {}
+{{- end }}
+---
+apiVersion: cert-manager.io/v1
+kind: Certificate
+metadata:
+  name: {{ include "spire-server.fullname" $ }}-ca
+  namespace: {{ include "spire-server.namespace" $ }}
+spec:
+  isCA: true
+  commonName: {{ $.Values.ca_subject.common_name }}
+  secretName: {{ include "spire-server.fullname" $ }}-ca-keys
+  privateKey:
+    algorithm: ECDSA
+    size: 256
```

Comment:
Thinking a lot more of this needs to be configurable too....
just glancing through
kubectl explain certificate.spec,

duration should be set-able. it defaults to 90 days.... thats probably way too short to be usable for anything other then testing. renewBefore and subject as perhaps too. We already take in subject config elsewhere.

privateKey has 4 settings under it... algorithm, encoding, rotationPolicy and size. All of those should maybe be configurable as well....

Reply:
A comment saying not to use this for production use should be sufficient.